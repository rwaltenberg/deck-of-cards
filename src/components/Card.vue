<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import { StringToCard } from '../utils/card-conversion'
import { SuitSymbols } from '../config/deck'

@Component
export default class DeckCard extends Vue {
  @Prop({ type: [Array, String], required: true })
  private value!: Card | string

  @Prop({ type: String, default: () => 'div' })
  private tag!: string

  get card (): Card {
    if (Array.isArray(this.value)) {
      return this.value.slice(0) as Card
    }

    return StringToCard(this.value)
  }

  getSymbol (suit: Suit) {
    return SuitSymbols[suit]
  }
}
</script>

<template>
  <component :is="tag" class="card">
    <div class="value">
      <div class="rank">{{ card[0] }}</div>
      <div class="suit">{{ getSymbol(card[1]) }}</div>
    </div>
    <div class="value mirror">
      <div class="rank">{{ card[0] }}</div>
      <div class="suit">{{ getSymbol(card[1]) }}</div>
    </div>
  </component>
</template>

<style lang="scss" scoped>
  .card {
    background-color: #ffffff;
    border-radius: 15px;
    box-shadow: 5px 5px 10px #9182a1;
    box-sizing: border-box;
    overflow: hidden;
    padding: 10px 20px;

    .value {
      align-items: flex-start;
      display: flex;
      font-family: Arial, serif;
      font-size: rem(32px);
      font-weight: bold;
      line-height: 1;
      height: 110px;
      width: 100px;

      &.mirror {
        transform: scale(-1);
      }

      .rank {
        line-height: rem(55px);
        margin-right: 5px;
      }

      .suit {
        font-size: rem(50px);
      }
    }
  }
</style>
