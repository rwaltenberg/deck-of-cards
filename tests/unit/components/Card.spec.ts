import Card from '@/components/Card.vue'
import { shallowMount } from '@vue/test-utils'

describe('Card.vue', () => {
  describe('with string type value', () => {
    it('renders properly', () => {
      const wrapper = shallowMount(Card, {
        propsData: {
          value: 'AS'
        }
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with Card type value', () => {
    it('renders properly', () => {
      const wrapper = shallowMount(Card, {
        propsData: {
          value: ['A', 'S']
        }
      })
      expect(wrapper).toMatchSnapshot()
    })
  })
})
