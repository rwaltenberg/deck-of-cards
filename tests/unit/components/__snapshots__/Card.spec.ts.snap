// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue with Card type value renders properly 1`] = `
<div class="card">
  <div class="value" style="color: rgb(0, 0, 0);">
    <div class="rank">A</div>
    <div class="suit">♠</div>
  </div>
  <div class="value mirror" style="color: rgb(0, 0, 0);">
    <div class="rank">A</div>
    <div class="suit">♠</div>
  </div>
</div>
`;

exports[`Card.vue with string type value renders properly 1`] = `
<div class="card">
  <div class="value" style="color: rgb(0, 0, 0);">
    <div class="rank">A</div>
    <div class="suit">♠</div>
  </div>
  <div class="value mirror" style="color: rgb(0, 0, 0);">
    <div class="rank">A</div>
    <div class="suit">♠</div>
  </div>
</div>
`;
